lms/djangoapps/django_comment_client/base/views.py

```diff
@@ -533,7 +533,7 @@ def _vote_or_unvote(request, course_id, obj, value='up', undo_vote=False):
         # (People could theoretically downvote by handcrafting AJAX requests.)
     else:
         user.vote(obj, value)
-    thread_voted.send(sender=None, user=request.user, post=obj)
+    thread_voted.send(sender=None, user=request.user, post=obj, undo_vote=undo_vote)
```

Comment:
@tehreem-sadat I think we should send a separate `thread_unvoted` signal in case of unvote.

Reply:
I can handle the unvote scenario with a custom "thread_unvote" signal. 

But the problem remains same for vote scenario as the edX "thread_vote" signal will be called for both vote and unvote cases and in my handler of thread_vote there is no way to check for which case (vote or unvote) signal is being generated as a result email will be sent for both cases. 

Instead of custom "thread_unvote" we can write our custom signal (that will handle both vote and unvote) to which we can send any extra parameters (undo flag) without changing any edx signals.